packages/@aws-cdk/aws-glue-alpha/lib/jobs/pyspark-flex-etl-job.ts

```diff
@@ -0,0 +1,197 @@
+    const observabilityMetricsArgs = { '--enable-observability-metrics': 'true' };
+
+    // Gather executable arguments
+    const execuatbleArgs = this.executableArguments(props);
+
+    // Conbine command line arguments into a single line item
+    const defaultArguments = {
+      ...execuatbleArgs,
+      ...continuousLoggingArgs,
+      ...profilingMetricsArgs,
+      ...observabilityMetricsArgs,
+      ...sparkUIArgs?.args,
+      ...this.checkNoReservedArgs(props.defaultArguments),
+    };
+
+    const jobResource = new CfnJob(this, 'Resource', {
+      name: props.jobName,
+      description: props.description,
+      role: this.role.roleArn,
+      command: {
+        name: JobType.ETL,
+        scriptLocation: this.codeS3ObjectUrl(props.script),
+        pythonVersion: PythonVersion.THREE,
+      },
+      glueVersion: props.glueVersion ? props.glueVersion : GlueVersion.V3_0,
+      workerType: props.workerType ? props.workerType : WorkerType.G_1X,
+      numberOfWorkers: props.numberOfWorkers ? props.numberOfWorkers : 10,
+      maxRetries: props.maxRetries,
+      executionProperty: props.maxConcurrentRuns ? { maxConcurrentRuns: props.maxConcurrentRuns } : undefined,
+      notificationProperty: props.notifyDelayAfter ? { notifyDelayAfter: props.notifyDelayAfter.toMinutes() } : undefined,
+      timeout: props.timeout?.toMinutes(),
+      connections: props.connections ? { connections: props.connections.map((connection) => connection.connectionName) } : undefined,
+      securityConfiguration: props.securityConfiguration?.securityConfigurationName,
+      tags: props.tags,
+      executionClass: ExecutionClass.FLEX,
+      defaultArguments,
+    });
+
+    const resourceName = this.getResourceNameAttribute(jobResource.ref);
+    this.jobArn = this.buildJobArn(this, resourceName);
+    this.jobName = resourceName;
+  }
+
+  /**
+   *Set the executable arguments with best practices enabled by default
+   *
+   * @param props
+   * @returns An array of arguments for Glue to use on execution
+   */
+  private executableArguments(props: PySparkFlexEtlJobProps) {
+    const args: { [key: string]: string } = {};
+    args['--job-language'] = JobLanguage.PYTHON;
+
+    if (props.extraPythonFiles && props.extraPythonFiles.length > 0) {
+      args['--extra-py-files'] = props.extraPythonFiles.map(code => this.codeS3ObjectUrl(code)).join(',');
+    }
+    if (props.extraFiles && props.extraFiles.length > 0) {
+      args['--extra-files'] = props.extraFiles.map(code => this.codeS3ObjectUrl(code)).join(',');
+    }
+
+    return args;
+  }
+
+  /**
+   * Set the arguments for sparkUI with best practices enabled by default
+   *
+   * @param sparkUiProps, role
+   * @returns An array of arguments for enabling sparkUI
+   */
+
+  private setupSparkUI(role: iam.IRole, sparkUiProps: SparkUIProps) {
+
+    validateSparkUiPrefix(sparkUiProps.prefix);
+    const bucket = sparkUiProps.bucket ?? new Bucket(this, 'SparkUIBucket');
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Bucket is not set to enforce encryption-in-transit, if not explictly setting this on the bucket policy - the property &#34;enforceSSL&#34; should be set to true.

Suggested remediation:
Bucket is not set to enforce encryption-in-transit, if not explictly setting this on the bucket policy - the property &#34;enforceSSL&#34; should be set to true.
```
@@ -180,5 +180,5 @@
 
     validateSparkUiPrefix(sparkUiProps.prefix);
-    const bucket = sparkUiProps.bucket ?? new Bucket(this, &#39;SparkUIBucket&#39;);
+    const bucket = sparkUiProps.bucket ?? new Bucket(this, &#39;SparkUIBucket&#39;, { enforceSSL: true } );
     bucket.grantReadWrite(role, cleanSparkUiPrefixForGrant(sparkUiProps.prefix));
     const args = {

```